### PR TITLE
Add Dynamic Partition Tools

### PR DESCRIPTION
**Pull request by MoetaYuko**
This PR adds tools to handle dynamic partitions (super.img), tested working on archlinux.

### CODE REVIEW

File: patches/extras/0002-libjsonpbparse-Fix-build-after-protobuf-util-Status-.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+diff --git a/libjsonpb/parse/jsonpb.cpp b/libjsonpb/parse/jsonpb.cpp
+index 3a042e7..e65a39d 100644
+--- a/libjsonpb/parse/jsonpb.cpp
++++ b/libjsonpb/parse/jsonpb.cpp
+@@ -48,7 +48,7 @@ ErrorOr<std::string> MessageToJsonString(const Message& message) {
+                                    &json, options);
+ 
+   if (!status.ok()) {
+-    return MakeError<std::string>(status.error_message().as_string());
++    return MakeError<std::string>(status.message().as_string());
```
Comment (MoetaYuko) [inline patches/extras/0002-libjsonpbparse-Fix-build-after-protobuf-util-Status-.patch:20]: I'm not sure if this needs to be ifdef'd. https://github.com/protocolbuffers/protobuf/pull/8354 is the breaking change.

**Comment by MoetaYuko** [comment]
BTW the library dependencies are not well-organized. Arguably each static/dynamic library should declare its dependencies via `target_link_libraries` and export its headers via `target_include_directories`, rather than handcrafting all deps when building the executable. Any plan to refactor in the future?

**Comment by nmeum** [comment]
> BTW the library dependencies are not well-organized. Arguably each static/dynamic library should declare its dependencies via `target_link_libraries` and export its headers via `target_include_directories`, rather than handcrafting all deps when building the executable. Any plan to refactor in the future?

Yes, that's currently a bit messy indeed. I personally haven't gotten around to refactoring this part of the build system yet and I am not sure if I will have the time do to so in the foreseeable future but I would welcome contributions addressing this issue.

**Comment by MoetaYuko** [comment]
@nmeum Now it passes the CI build in my fork, please double-check.

**Comment by MoetaYuko** [comment]
bump, is this mergeable now?

**Pull request merged**